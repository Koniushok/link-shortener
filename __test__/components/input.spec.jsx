import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import Input, { LabelError } from '../../src/components/input';

describe('Input component', () => {
  const props = {
    defaultValue: 'default',
    onChange: () => undefined,
    name: 'testName',
    type: 'password',
    label: 'testLabel',
    error: 'testError',
  };

  describe('Input component initial', () => {
    it('render correctly component', () => {
      const output = shallow(<Input {...props} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });

  describe('Input component render <LabelError>', () => {
    it('render error', () => {
      const input = mount(<Input {...props} />);
      expect(input.find(LabelError)).toHaveLength(1);
    });
    it('not render error', () => {
      const input = mount(<Input name="test" label="label" />);
      expect(input.find(LabelError)).toHaveLength(0);
    });
  });
});
