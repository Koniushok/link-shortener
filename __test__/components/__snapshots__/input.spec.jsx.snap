// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input component initial render correctly component 1`] = `
<div>
  <styled.label
    htmlFor="testName"
  >
    testLabel
  </styled.label>
  <styled.input
    autocomplete="off"
    defaultValue="default"
    error={true}
    id="testName"
    name="testName"
    onChange={[Function]}
    type="password"
  />
  <styled.div>
    testError
  </styled.div>
</div>
`;
